spec/models/generic_file_spec.rb

```diff
@@ -625,4 +625,156 @@
       end
     end
   end
+
+  let(:public_permission_double) { double("Hydra::AccessControls::Permission") }
```

Comment:
[Here is the documentation ](https://relishapp.com/rspec/rspec-mocks/docs/basics/test-doubles). As I understand it, it's an object from the Rspec library that you can send messages (function calls) to and receive pre-set return values. I think the thing that separates it from a mock or spy (they might not be a part of rspec anymore, but were in the past) is that it will verify the object you passed as a string actually has the method you specified to mock out.